### PR TITLE
Add check for ssh port 22

### LINKED ISSUE #70: Enhancement: Don't expose SSH ports
A check we should consider for KubeLinter is not to expose ssh ports as part of a deployment. @viswajithiii lmk your thoughts here.

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

Fixes #70

### CODE REVIEW

File: internal/templates/port/template.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/port/internal/params"
+	"golang.stackrox.io/kube-linter/internal/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "ports",
```
Comment (kreamkorokke) [inline internal/templates/port/template.go:17]: nit: Ports?

File: internal/builtinchecks/yamls/ssh-port.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: "ssh-port"
+description: "Alert on deployments with containers exposing SSH port 22"
+remediation: "Don't expose SSH port as it's not recommended as per security best practice."
+scope:
+  objectKinds:
+    - DeploymentLike
+template: "port"
+params:
+  port: 22
+  protocol: "TCP"
```
Comment (kreamkorokke) [inline internal/builtinchecks/yamls/ssh-port.yaml:10]: nit: new line at the end of the file

File: internal/templates/port/template.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		Key:         "ports",
+		Description: "Flag containers exposing ports which not recommended as per security best practice",
```
Comment (kreamkorokke) [inline internal/templates/port/template.go:19]: maybe change description to: `Flag containers exposing ports under protocols that match the supplied parameters`

File: internal/templates/port/template.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+				var results []diagnostic.Diagnostic
+				for _, port := range container.Ports {
+					if int(port.ContainerPort) == p.Port && string(port.Protocol) == p.Protocol {
```
Comment (kreamkorokke) [inline internal/templates/port/template.go:29]: for matching protocol, please use `matcher.ForString`

a usage example: https://github.com/stackrox/kube-linter/blob/main/internal/templates/envvar/template.go#L28 :)

File: internal/builtinchecks/yamls/ssh-port.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+name: "ssh-port"
+description: "Alert on deployments with containers exposing SSH port 22"
```
Comment (kreamkorokke) [inline internal/builtinchecks/yamls/ssh-port.yaml:2]: nit: change to `Alert on deployments exposing port 22, commonly reserved for SSH access.`

File: internal/builtinchecks/yamls/ssh-port.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+remediation: "Don't expose SSH port as it's not recommended as per security best practice."
```
Comment (kreamkorokke) [inline internal/builtinchecks/yamls/ssh-port.yaml:3]: let's be a bit more concrete with the remediation, something like: `Ensure that non-SSH services are not using port 22. Ensure that any actual SSH servers have been vetted.`

**Review by kreamkorokke — COMMENTED** [review]
Hey @arghya88 thank you very much again for submitting PR! Left several comments but overall looks good to me!

**Comment by arghya88** [comment]
@kreamkorokke done the changes.PTAL

**Review by kreamkorokke — APPROVED** [review]
Looks good! Sorry for the delay. Merging now

**Pull request merged**